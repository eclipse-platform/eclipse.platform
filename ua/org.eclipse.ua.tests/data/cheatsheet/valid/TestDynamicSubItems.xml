<?xml version="1.0" encoding="UTF-8" ?> 
<!--
     Copyright (c) 2005, 2006 IBM Corporation and others.

     This program and the accompanying materials
     are made available under the terms of the Eclipse Public License 2.0
     which accompanies this distribution, and is available at
     https://www.eclipse.org/legal/epl-2.0/

     SPDX-License-Identifier: EPL-2.0
    
     Contributors:
         IBM Corporation - initial API and implementation
 -->

<cheatsheet title="Testing SubItems">

<intro 
	href="/org.eclipse.platform.doc.user/reference/ref-cheatsheets.htm">
	<description>
This cheat sheet will test subitems.
Let's get started!
	</description>
</intro>

	<item
		title="Select a pattern">
		<action
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ui.examples.cheatsheets.pattern.actions.LaunchPatternSelectionDialogAction"/>
		<description>
Launch the wizard to begin. You can select a pattern to create. There
are a few different patterns that you can try out, follow the steps in the
cheat sheet to help guide you through building the patterns.
		</description>
</item>

<!--
Run the cheat once with the following 2 items commented out. Work through the cheat sheet
until the second "Customize your pattern" item. Exit the cheat sheet and uncomment the items
to test the error page during the restoration of saved state data.
-->
<!--
	<item
		title="Test conditional-subitem">
		<description>
You should see the "Step for A." subitem.
		</description>
		<conditional-subitem condition="a">
			<subitem when="a" label="Step for A." />
			<subitem when="b" label="Step for B." />
		</conditional-subitem>
	</item>

	<item
		title="Test repeated-subitem">
		<description>
You should see the "Step for A.", "Step for B." and "Step for C." subitems.
		</description>
		<repeated-subitem values="A,B,C">
			<subitem label="Step for ${this}"/>
		</repeated-subitem>
	</item>
-->
	<item
		title="Customize your pattern">
		<description>
Now that you have chosen the pattern that you would like to create, customize your options.
		</description>
	<conditional-subitem condition="${pattern}">
		<subitem
			when="Factory"
			label="Customize the Factory Pattern"
			skip="true">
			<action
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
		</subitem>
		<subitem
			when="Singleton"
			label="Customize the Singleton Pattern"
			skip="false">
			<action
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
		</subitem>
		<subitem
			when="Visitor"
			label="Customize the Visitor Pattern"
			skip="true">
			<action
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</subitem>
	</conditional-subitem>
	</item>

<item
	title="Test SubItems"
	href="/org.eclipse.platform.doc.user/concepts/chelpsys.htm">
	<description>
Try the help link, you should see the concepts for the help system.
	</description>
	<subitem label="sub 1" skip="true"/>
	<subitem label="sub 2" skip="true">
		<action
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<conditional-subitem condition="${pattern}">
		<subitem
			when="Factory"
			label="Customize the Factory Pattern"
			skip="false">
			<action
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
		</subitem>
		<subitem
			when="Singleton"
			label="Customize the Singleton Pattern"
			skip="false">
			<action
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
		</subitem>
		<subitem
			when="Visitor"
			label="Customize the Visitor Pattern"
			skip="true">
			<action
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</subitem>
	</conditional-subitem>
	<conditional-subitem condition="${pattern}">
		<subitem
			when="Factory"
			label="Customize the Factory Pattern">
			<action
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
		</subitem>
		<subitem
			when="Singleton"
			label="Customize the Singleton Pattern">
			<action
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
		</subitem>
		<subitem
			when="Visitor"
			label="Customize the Visitor Pattern">
		</subitem>
	</conditional-subitem>
	<conditional-subitem condition="${pattern}">
		<subitem
			when="Factory"
			label="Customize the Factory Pattern"
			skip="true">
		</subitem>
		<subitem
			when="Singleton"
			label="Customize the Singleton Pattern"
			skip="false">
		</subitem>
		<subitem
			when="Visitor"
			label="Customize the Visitor Pattern">
		</subitem>
	</conditional-subitem>
	<subitem label="sub 3" skip="false"/>
	<subitem label="sub 4" skip="false">
		<action
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 5"/>
	<subitem label="sub 6">
		<action
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 7">
		<action
			confirm="true"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 8">
		<action
			confirm="false"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 9" skip="true">
		<action
			confirm="true"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 10" skip="true">
		<action
			confirm="false"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<repeated-subitem values="${files}">
		<subitem label="Create class ${this}">
			<action
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.OpenClassNameFilledAction"
				param1="${this}"
				param2="${project}"/>
		</subitem>
	</repeated-subitem>
	<subitem label="sub 11" skip="false">
		<action
			confirm="false"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
	<subitem label="sub 12" skip="false">
		<action
			confirm="true"
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"/>
	</subitem>
</item>

	<item
		title="Select a pattern">
		<action
			pluginId="org.eclipse.ua.tests"
			class="org.eclipse.ui.examples.cheatsheets.pattern.actions.LaunchPatternSelectionDialogAction"/>
		<description>
Launch the wizard to begin. You can select a pattern to create. There
are a few different patterns that you can try out, follow the steps in the
cheat sheet to help guide you through building the patterns.
		</description>
</item>

<item
	title="Test SubItems with PerformWhen"
	href="/org.eclipse.platform.doc.user/concepts/chelpsys.htm">
	<description>
Try the help link, you should see the concepts for the help system.
	</description>
	<subitem label="sub 1">
		<perform-when condition="${pattern}">
			<action
				when="Factory"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
			<action
				when="Singleton"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
			<action
				when="Visitor"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</perform-when>
	</subitem>
	<subitem label="sub 2" skip="false">
		<perform-when condition="${pattern}">
			<action
				when="Factory"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
			<action
				when="Singleton"
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
			<action
				when="Visitor"
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</perform-when>
	</subitem>
	<subitem label="sub 3" skip="true">
		<perform-when condition="${pattern}">
			<action
				when="Factory"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
			<action
				when="Singleton"
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
			<action
				when="Visitor"
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</perform-when>
	</subitem>
</item>

	<item
		title="Customize your pattern">
		<description>
Now that you have chosen the pattern that you would like to create, customize your options.
		</description>
	<conditional-subitem condition="${pattern}">
		<subitem
			when="Factory"
			label="Customize the Factory Pattern"
			skip="true">
			<perform-when condition="${pattern}">
				<!-- The "when"  are set incorrectly to ensure the perform when is working. -->
				<action
					when="Singleton"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
				<action
					when="Visitor"
					confirm="false"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
				<action
					when="Factory"
					confirm="true"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
			</perform-when>
		</subitem>
		<subitem
			when="Singleton"
			label="Customize the Singleton Pattern"
			skip="false">
			<action
				confirm="true"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
		</subitem>
		<subitem
			when="Visitor"
			label="Customize the Visitor Pattern"
			skip="true">
			<action
				confirm="false"
				pluginId="org.eclipse.ua.tests"
				class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
		</subitem>
	</conditional-subitem>
	</item>

	<item
		title="Customize your pattern">
		<description>
Now that you have chosen the pattern that you would like to create, customize your options.
		</description>
	<repeated-subitem values="${files}">
		<subitem label="Create class ${this}">
			<perform-when condition="${pattern}">
				<!-- The "when"  are set incorrectly to ensure the perform when is working. -->
				<action
					when="Singleton"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"/>
				<action
					when="Visitor"
					confirm="false"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"/>
				<action
					when="Factory"
					confirm="true"
					pluginId="org.eclipse.ua.tests"
					class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"/>
			</perform-when>
		</subitem>
	</repeated-subitem>
	</item>

<item
	title="All done!">
	<description>
Did the test work?
	</description>
</item>

</cheatsheet>
