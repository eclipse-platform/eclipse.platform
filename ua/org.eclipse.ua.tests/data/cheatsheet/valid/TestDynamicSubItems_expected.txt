<?xml version="1.0" encoding="UTF-8" ?>
<cheatsheet
      title="Testing SubItems">
   <item
         title="Introduction"
         description="This cheat sheet will test subitems.
Let's get started!"
         Href="/org.eclipse.platform.doc.user/reference/ref-cheatsheets.htm"
         contextId="null">
      <nullAction/>
      <nullList/>
      <nullPerformWhen/>
      <nullList/>
   </item>
   <list>
      <item
            title="Select a pattern"
            description="Launch the wizard to begin. You can select a pattern to create. There
are a few different patterns that you can try out, follow the steps in the
cheat sheet to help guide you through building the patterns."
            Href="null"
            contextId="null">
         <action
               class="org.eclipse.ui.examples.cheatsheets.pattern.actions.LaunchPatternSelectionDialogAction"
               pluginId="org.eclipse.ua.tests"
               when="null"
               isConfirm="false">
            <nullArray/>
         </action>
         <list/>
         <nullPerformWhen/>
         <nullList/>
      </item>
      <item
            title="Customize your pattern"
            description="Now that you have chosen the pattern that you would like to create, customize your options."
            Href="null"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <list>
            <conditionalSubItem
                  condition="${pattern}">
               <list>
                  <subItem
                        label="Customize the Factory Pattern"
                        when="Factory"
                        isSkip="true">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Singleton Pattern"
                        when="Singleton"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Visitor Pattern"
                        when="Visitor"
                        isSkip="true">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </conditionalSubItem>
         </list>
      </item>
      <item
            title="Test SubItems"
            description="Try the help link, you should see the concepts for the help system."
            Href="/org.eclipse.platform.doc.user/concepts/chelpsys.htm"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <list>
            <subItem
                  label="sub 1"
                  when="null"
                  isSkip="true">
               <nullAction/>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 2"
                  when="null"
                  isSkip="true">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <conditionalSubItem
                  condition="${pattern}">
               <list>
                  <subItem
                        label="Customize the Factory Pattern"
                        when="Factory"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Singleton Pattern"
                        when="Singleton"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Visitor Pattern"
                        when="Visitor"
                        isSkip="true">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </conditionalSubItem>
            <conditionalSubItem
                  condition="${pattern}">
               <list>
                  <subItem
                        label="Customize the Factory Pattern"
                        when="Factory"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Singleton Pattern"
                        when="Singleton"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Visitor Pattern"
                        when="Visitor"
                        isSkip="false">
                     <nullAction/>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </conditionalSubItem>
            <conditionalSubItem
                  condition="${pattern}">
               <list>
                  <subItem
                        label="Customize the Factory Pattern"
                        when="Factory"
                        isSkip="true">
                     <nullAction/>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Singleton Pattern"
                        when="Singleton"
                        isSkip="false">
                     <nullAction/>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Visitor Pattern"
                        when="Visitor"
                        isSkip="false">
                     <nullAction/>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </conditionalSubItem>
            <subItem
                  label="sub 3"
                  when="null"
                  isSkip="false">
               <nullAction/>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 4"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 5"
                  when="null"
                  isSkip="false">
               <nullAction/>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 6"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 7"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="true">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 8"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 9"
                  when="null"
                  isSkip="true">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="true">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 10"
                  when="null"
                  isSkip="true">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <repeatedSubItem
                  values="${files}">
               <list>
                  <subItem
                        label="Create class ${this}"
                        when="null"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.OpenClassNameFilledAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <array>
                           <string
                                 value="${this}">
                           </string>
                           <string
                                 value="${project}">
                           </string>
                           <nullObject/>
                           <nullObject/>
                           <nullObject/>
                           <nullObject/>
                           <nullObject/>
                           <nullObject/>
                           <nullObject/>
                        </array>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </repeatedSubItem>
            <subItem
                  label="sub 11"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="false">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
            <subItem
                  label="sub 12"
                  when="null"
                  isSkip="false">
               <action
                     class="org.eclipse.ua.tests.cheatsheet.util.NoopAction"
                     pluginId="org.eclipse.ua.tests"
                     when="null"
                     isConfirm="true">
                  <nullArray/>
               </action>
               <nullPerformWhen/>
            </subItem>
         </list>
      </item>
      <item
            title="Select a pattern"
            description="Launch the wizard to begin. You can select a pattern to create. There
are a few different patterns that you can try out, follow the steps in the
cheat sheet to help guide you through building the patterns."
            Href="null"
            contextId="null">
         <action
               class="org.eclipse.ui.examples.cheatsheets.pattern.actions.LaunchPatternSelectionDialogAction"
               pluginId="org.eclipse.ua.tests"
               when="null"
               isConfirm="false">
            <nullArray/>
         </action>
         <list/>
         <nullPerformWhen/>
         <nullList/>
      </item>
      <item
            title="Test SubItems with PerformWhen"
            description="Try the help link, you should see the concepts for the help system."
            Href="/org.eclipse.platform.doc.user/concepts/chelpsys.htm"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <list>
            <subItem
                  label="sub 1"
                  when="null"
                  isSkip="false">
               <nullAction/>
               <performWhen
                     condition="${pattern}">
                  <list>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Factory"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Singleton"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Visitor"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                  </list>
               </performWhen>
            </subItem>
            <subItem
                  label="sub 2"
                  when="null"
                  isSkip="false">
               <nullAction/>
               <performWhen
                     condition="${pattern}">
                  <list>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Factory"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Singleton"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Visitor"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                  </list>
               </performWhen>
            </subItem>
            <subItem
                  label="sub 3"
                  when="null"
                  isSkip="true">
               <nullAction/>
               <performWhen
                     condition="${pattern}">
                  <list>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Factory"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Singleton"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="Visitor"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                  </list>
               </performWhen>
            </subItem>
         </list>
      </item>
      <item
            title="Customize your pattern"
            description="Now that you have chosen the pattern that you would like to create, customize your options."
            Href="null"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <list>
            <conditionalSubItem
                  condition="${pattern}">
               <list>
                  <subItem
                        label="Customize the Factory Pattern"
                        when="Factory"
                        isSkip="true">
                     <nullAction/>
                     <performWhen
                           condition="${pattern}">
                        <list>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Singleton"
                                 isConfirm="false">
                              <nullArray/>
                           </action>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Visitor"
                                 isConfirm="false">
                              <nullArray/>
                           </action>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Factory"
                                 isConfirm="true">
                              <nullArray/>
                           </action>
                        </list>
                     </performWhen>
                  </subItem>
                  <subItem
                        label="Customize the Singleton Pattern"
                        when="Singleton"
                        isSkip="false">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="true">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
                  <subItem
                        label="Customize the Visitor Pattern"
                        when="Visitor"
                        isSkip="true">
                     <action
                           class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                           pluginId="org.eclipse.ua.tests"
                           when="null"
                           isConfirm="false">
                        <nullArray/>
                     </action>
                     <nullPerformWhen/>
                  </subItem>
               </list>
            </conditionalSubItem>
         </list>
      </item>
      <item
            title="Customize your pattern"
            description="Now that you have chosen the pattern that you would like to create, customize your options."
            Href="null"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <list>
            <repeatedSubItem
                  values="${files}">
               <list>
                  <subItem
                        label="Create class ${this}"
                        when="null"
                        isSkip="false">
                     <nullAction/>
                     <performWhen
                           condition="${pattern}">
                        <list>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.FactoryCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Singleton"
                                 isConfirm="false">
                              <nullArray/>
                           </action>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.SingletonCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Visitor"
                                 isConfirm="false">
                              <nullArray/>
                           </action>
                           <action
                                 class="org.eclipse.ui.examples.cheatsheets.pattern.actions.VisitorCustomizeAction"
                                 pluginId="org.eclipse.ua.tests"
                                 when="Factory"
                                 isConfirm="true">
                              <nullArray/>
                           </action>
                        </list>
                     </performWhen>
                  </subItem>
               </list>
            </repeatedSubItem>
         </list>
      </item>
      <item
            title="All done!"
            description="Did the test work?"
            Href="null"
            contextId="null">
         <nullAction/>
         <list/>
         <nullPerformWhen/>
         <nullList/>
      </item>
   </list>
</cheatsheet>
